tools/code-owners-parser/CodeOwnersParser/TeamUserHolder.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.CodeOwnersParser
+{
+    public class TeamUserHolder
+    {
+        private string TeamUserStorageURI { get; set; } = DefaultStorageConstants.DefaultStorageURI;
+        private Dictionary<string, List<string>>? _teamUserDict = null;
+
+        public Dictionary<string, List<string>> TeamUserDict
+        {
+            get
+            {
+                if (_teamUserDict == null)
+                {
+                    _teamUserDict = GetTeamUserData();
+                }
+                return _teamUserDict;
+            }
+            set
+            {
+                _teamUserDict = value;
+            }
+        }
+
+        public TeamUserHolder(string? teamUserStorageURI)
+        {
+            if (!string.IsNullOrWhiteSpace(teamUserStorageURI))
+            {
+                TeamUserStorageURI = teamUserStorageURI;
+            }
+        }
+
+        private Dictionary<string, List<string>> GetTeamUserData()
+        {
+            if (null == _teamUserDict)
+            {
+                string rawJson = FileHelpers.GetFileOrUrlContents(TeamUserStorageURI);
+                var list = JsonSerializer.Deserialize<List<KeyValuePair<string, List<string>>>>(rawJson);
+                if (null != list)
+                {
+                    return list.ToDictionary((keyItem) => keyItem.Key, (valueItem) => valueItem.Value);
+                }
+                Console.WriteLine($"Error! Unable to deserialize json team/user data. rawJson={rawJson}");
+                return new Dictionary<string, List<string>>();
+            }
+            return _teamUserDict;
+        }
+
+        public List<string> GetUsersForTeam(string teamName)
+        {
+            // The teamName in the codeowners file should be in the form <org>/<team>.
+            // The dictionary's team names do not contain the org so the org needs to
+            // be stripped off. Handle the case where the teamName passed in does and
+            // does not being with @org/
+            string teamWithoutOrg = teamName.Trim();
+            if (teamWithoutOrg.Contains('/'))
+            {
+                teamWithoutOrg = teamWithoutOrg.Split("/")[1];
+            }
+            if (TeamUserDict != null)
+            {
+                if (TeamUserDict.ContainsKey(teamWithoutOrg))
+                {
+                    Console.WriteLine($"Found team entry for {teamWithoutOrg}");
```

Comment:
We might want to think about making this logging configurable or use a logging interface at some point as generally it isn't great for shared libraries to have a bunch of logging.

Reply:
That won't be part of this PR. I created a [new issue](https://github.com/Azure/azure-sdk-tools/issues/6438) to track this.